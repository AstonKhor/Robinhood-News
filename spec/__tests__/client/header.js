import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Header from '../../../client/components/header.jsx';


describe('Header Component', () => {
  it('should render with no problems', () => {
    // TODO finish when buidling  frontend
    const component = renderer.create(<Header />)
    expect(component).toMatchSnapshot();
  })

  it('should take in a prop for article titles as the header', () => {
    // fill in the enzyme/ react renderer create after the component is built
    expect(component.props().articles).to.be.lengthOf(3);
    expect(component.props().articles[0].title).toBeDefined();
  })

  it('should be bolded', () => {
    // fill in the enzyme/ react renderer create after the component is built
    expect(component.find('.title').hasClass('bold')).to.equal(true)
  })
})
