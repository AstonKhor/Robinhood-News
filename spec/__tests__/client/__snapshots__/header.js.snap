// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render with no problems 1`] = `
<div
  className="sc-bZQynM evQXcO"
>
  <span
    className="bold"
  >
    <span
      className="sc-gzVnrw gftquo"
    >
      Benzinga
    </span>
  </span>
  <span
    className="sc-htoDjs gbfbMp"
  >
    Dec 2
  </span>
</div>
`;
